Version 12: `java.lang.NoClassDefFoundError` thrown by `new DefaultUnleash(unleashConfig)`

### Describe the bug

Hi Unleash team! After upgrading to version 12 I got this error when calling `new DefaultUnleash(unleashConfig)`:

> java.lang.ClassNotFoundException: com.launchdarkly.eventsource.ConnectStrategy

I believe this is because `com.launchdarkly.okhttp-eventsource` is marked as `optional` in `pom.xml`:

https://github.com/Unleash/unleash-java-sdk/blob/ecfd93bc0c467a53ba6ed0cb7f2b6cc27d04e302/pom.xml#L77-L82

On further investigation, it seems `okhttp-eventsource` was added to support streaming in #318. At the time the class was only used if `unleashConfig.experimentalMode` was set to `STREAMING`.

In #335 the code was changed to create the `StreamingFeatureFetcherImpl` regardless of the value of `experimentalMode`, thus the `ClassNotFoundException` if I haven't added the dependency myself.

### Steps to reproduce the bug

```java
@Bean
public Unleash unleash() {
    final var config = UnleashConfig.builder()
            .appName(appName)
            .instanceId(instanceId)
            .unleashAPI(apiUrl)
            .synchronousFetchOnInitialisation(true)
            .build();

    return new DefaultUnleash(config);
}
```

### Expected behavior

I may be misunderstanding, but it seems like `com.launchdarkly.okhttp-eventsource` is no longer truly optional. Should we just remove `<optional>true</optional>` from the pom file? Or should the code be adjusted to only create the `StreamingFeatureFetcherImpl` if `experimentalMode` is set to `STREAMING`?

Thanks in advance for any advice!

### Logs, error output, etc.

```bash
Caused by: java.lang.NoClassDefFoundError: com/launchdarkly/eventsource/ConnectStrategy
	at io.getunleash.repository.AdaptiveFetcher.defaultWorkers(AdaptiveFetcher.java:43) ~[unleash-client-java-12.0.0.jar:12.0.0]
	at io.getunleash.repository.AdaptiveFetcher.<init>(AdaptiveFetcher.java:63) ~[unleash-client-java-12.0.0.jar:12.0.0]
	at io.getunleash.repository.AdaptiveFetcher.<init>(AdaptiveFetcher.java:54) ~[unleash-client-java-12.0.0.jar:12.0.0]
	at io.getunleash.repository.FeatureRepositoryImpl.<init>(FeatureRepositoryImpl.java:39) ~[unleash-client-java-12.0.0.jar:12.0.0]
	at io.getunleash.repository.FeatureRepositoryImpl.<init>(FeatureRepositoryImpl.java:28) ~[unleash-client-java-12.0.0.jar:12.0.0]
	at io.getunleash.EngineProxyImpl.<init>(EngineProxyImpl.java:38) ~[unleash-client-java-12.0.0.jar:12.0.0]
	at io.getunleash.DefaultUnleash.defaultToggleRepository(DefaultUnleash.java:42) ~[unleash-client-java-12.0.0.jar:12.0.0]
	at io.getunleash.DefaultUnleash.<init>(DefaultUnleash.java:46) ~[unleash-client-java-12.0.0.jar:12.0.0]
```

### Screenshots

_No response_

### Additional context

_No response_

### Unleash version

12

### Subscription type

None

### Hosting type

None

### SDK information (language and version)

_No response_

--- a/pom.xml
+++ b/pom.xml
@@ -78,7 +78,6 @@
             <groupId>com.launchdarkly</groupId>
             <artifactId>okhttp-eventsource</artifactId>
             <version>4.1.1</version>
-            <optional>true</optional>
         </dependency>
 
         <dependency>

